javascript/javascript1/week3/cactus_io.js

```diff
@@ -0,0 +1,128 @@
+const smartPhoneActivities = [];
+const usageLimit = 120;
+
+function addActivity(activity, duration, activities) {
+    const errorMessages = [];
+    const errorMessage = [
+        "ACTIVITY MUST BE A STRING!",
+        "DURATION MUST BE A NUMBER!",
+    ]
+
+    if (typeof activity !== "string") {
+        errorMessages.push(errorMessage[0]);
+    } 
+    
+    if (typeof duration !== "number") {
+        errorMessages.push(errorMessage[1]);
+    } 
+    
+    if (errorMessages.length > 0) {
+        console.log(errorMessages);
+        return;
+    } else {
+        activities.push({
+            date: new Date().toLocaleDateString("en-GB"),
+            activity: activity,
+            duration: duration
+        })
+    }
+}
+
+function timeSpent(activities) {
+    const today = new Date ();
+    const todayDate = today.toLocaleDateString("en-GB");
+
+    let totalDuration = 0;
+    for (let i = 0; i<activities.length; i++) {
+        if (activities[i]["date"] === todayDate) {
```

Comment:
We barely ever access key-values from an object using this syntax. The most common syntax is:
```suggestion
        if (activities[i].date === todayDate) {
```

Reply:
Thanks! Even though I've consulted the internet a million times about when to use dot notation or bracket notation (with and without quotes) to access a key value, I still struggle when I'm in action. Any tips to remember this more easily?